### ISSUE TITLE
[Bug]: At fresh install on Vue3 + Vite can't build storybook because TypeError: Cannot read properties of undefined (reading 'compilation') 

### ISSUE THREAD

**Issue by nuriaheep** [storybookjs/storybook#20183]
### Describe the bug

I did a fresh install of Vue 3 with Vite (with `npm init vue@latest`), added Storybook with `npx storybook init --type vue3` and when I do the command `npm run storybook` I get this error:

```
info @storybook/vue3 v6.5.14
info 
info => Loading presets
unable to close vite server
/Users/.../node_modules/@storybook/core-server/dist/cjs/build-dev.js:191
    } else if ((_error$compilation = error.compilation) !== null && _error$compilation !== void 0 && _error$compilation.errors) {
                                           ^

TypeError: Cannot read properties of undefined (reading 'compilation')
    at buildDev (/Users/.../node_modules/@storybook/core-server/dist/cjs/build-dev.js:191:44)

```
I'm using Node.js v18.12.1
My package.json:

````
{
  "name": "project-commons",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "run-p type-check build-only",
    "preview": "vite preview",
    "test:unit": "vitest --environment jsdom --root src/",
    "test:e2e": "playwright test",
    "build-only": "vite build",
    "type-check": "vue-tsc --noEmit -p tsconfig.vitest.json --composite false",
    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "vue": "^3.2.45"
  },
  "devDependencies": {
    "@babel/core": "^7.20.5",
    "@playwright/test": "^1.28.1",
    "@rushstack/eslint-patch": "^1.1.4",
    "@storybook/addon-actions": "^6.5.14",
    "@storybook/addon-essentials": "^6.5.14",
    "@storybook/addon-interactions": "^6.5.14",
    "@storybook/addon-links": "^6.5.14",
    "@storybook/builder-vite": "^0.2.5",
    "@storybook/testing-library": "^0.0.13",
    "@storybook/vue3": "^6.5.14",
    "@types/jsdom": "^20.0.1",
    "@types/node": "^18.11.9",
    "@vitejs/plugin-vue": "^3.2.0",
    "@vue/eslint-config-prettier": "^7.0.0",
    "@vue/eslint-config-typescript": "^11.0.0",
    "@vue/test-utils": "^2.2.4",
    "@vue/tsconfig": "^0.1.3",
    "babel-loader": "^8.3.0",
    "eslint": "^8.22.0",
    "eslint-plugin-storybook": "^0.6.8",
    "eslint-plugin-vue": "^9.3.0",
    "jsdom": "^20.0.3",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.7.1",
    "typescript": "~4.7.4",
    "vite": "^3.2.4",
    "vitest": "^0.25.3",
    "vue-loader": "^16.8.3",
    "vue-tsc": "^1.0.9"
  }
}
```




### To Reproduce

```shell
npm init vue@latest
npm install
npm run lint

npx storybook init --type vue3
run the 'eslintPlugin' migration? yes
run the 'npm7' migration? yes
npm run storybook
```


### System

_No response_

### Additional context

_No response_

**Comment by shilman** [comment]
Please try upgrading to the latest prerelease:

```
npx storybook@next upgrade --prerelease
```

Does that fix it?

**Comment by nuriaheep** [comment]
It works now!
Thank you

**Comment by thedamon** [comment]
I had similar problem and ran the command above which seemed to fix it.. however I didn't get any changes to my package.json or yarn.lock so i'm not sure.. still on @storybook/vue3 6.5.15. is that expected?

**Comment by sarahxoxo** [comment]
TL;DR: there're some npm configs storybook doesn't like it. Not sure how the entries got in npm_config and also why it only bothered storybook

I had a similar problem and ran the command above didn't fix it cuz it tried to upgrade to the latest version. 
Seems like an edge case, since it's only happening in one of our dev machines. 

We fixed it by the following steps:
1. Check your npm config by running `npm config list -l`
2. Look for any entries related to errors, if there's one just remove it
3. Restart your terminal and retry running your Storybook command
